ts/index.ts

```diff
@@ -0,0 +1,193 @@
+  private gitRoot: string;
+  private spdxExpression: string;
+  private errorOnConflict: boolean;
+  private includeOutputPaths: boolean;
+  private rolling: boolean;
+  private mirror: boolean;
+  private name: string | null;
+  private rollingMinor: number | null;
+
+  constructor() {
+    const options: ActionOptions = {
+      name: "flakehub-push",
+      fetchStyle: "gh-env-style",
+      diagnosticsUrl: new URL(
+        "https://install.determinate.systems/flakehub-push/telemetry",
+      ),
+      legacySourcePrefix: "flakehub-push",
+      requireNix: "fail",
+    };
+
+    this.idslib = new IdsToolbox(options);
+
+    // Inputs translated into environment variables for flakehub-push
+    this.tag = inputs.getString("tag");
+    this.host = inputs.getString("host");
+    this.logDirectives = inputs.getString("log-directives");
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    this.repository = inputs.getString("repository");
+    this.directory = inputs.getString("directory");
+    this.gitRoot = inputs.getString("git-root");
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.rolling = inputs.getBool("rolling");
+    this.mirror = inputs.getBool("mirror");
+    this.name = inputs.getStringOrNull("name");
+    this.rollingMinor = inputs.getNumberOrNull("rolling-minor");
+  }
+
+  // extra-tags is deprecated but we still honor it
+  private get extraLabels(): string {
+    const labels = inputs.getString("extra-labels"); // current input name
+    const tags = inputs.getString("extra-tags"); // deprecated input name
+
+    // If `extra-labels` is set to something use it, otherwise use `extra-tags`.
+    // It `extra-tags` is also not set, which means that it's an empty string, that's
+    // still valid, as the flakehub-push CLI expects a comma-separated list here.
+    return labels !== "" ? labels : tags;
+  }
+
+  // We first check for a value using the `source-binary` input and fall back to the
+  // now-deprecated `flakehub-push-binary`
+  private get sourceBinary(): string | null {
+    const sourceBinaryInput = inputs.getStringOrNull("source-binary");
+    const flakeHubPushBinaryInput = inputs.getStringOrNull(
+      "flakehub-push-binary",
+    );
+
+    return sourceBinaryInput !== ""
+      ? sourceBinaryInput
+      : flakeHubPushBinaryInput;
+  }
+
+  private get visibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      actionsCore.setFailed(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.map((opt) => `\`${opt}\``).join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
```

Comment:
Not well-verse in JS/TS -- does this need to be async/`Promise`? It doesn't do any async computation as far as I can see.

Reply:
Good call